### ISSUE TITLE
Consider dropping using STL Mersenne Twister PRNG

### ISSUE THREAD

**Issue by WojciechMigda** [WojciechMigda/Tsetlini#16]

**Comment by peteroupc** [comment]
Can you explain why you raised this issue?

**Comment by WojciechMigda** [comment]
There already is vectorized local implementation Mersenne Twister and I decided to simplify the code.

At the time when I was working on this piece over a year ago I was trying to make a case for either using `stl::mt19937` or my own `IRNG`. Running simple benchmarks from `examples` folder seemed to show that with `stl::mt19937` they ran faster so I stayed with stl.

This time when I got back to the project I shifted focus to bigger and longer running examples, which are bound by the most time-complex `calculate_clause_output` functionality anyway. So when this time I ran `mnist-2` example it didn't show any reasons to keep using stl PRNG here and I switched to IRNG of mine.

Also, in the longer timeframe this implementation of classic Tsetlin Machine algo will be superceded by the newer versions that were developed at CAIR which are more efficient and which I will hopefuly port. I am no longer putting that much attention to time efficiency of this classic implementation (great if it stays the same after the refactorings I already did and I plan to do) but my preference now goes to making sure it runs correctly and is easier to read & maintain. In the end I would like it to become a reference for my future development within this project.
